app/jobs/mfa_usage_stats_job.rb

```diff
@@ -0,0 +1,15 @@
+class MfaUsageStatsJob < ApplicationJob
+  queue_as :default
+
+  def perform
+    non_mfa_users = User.where(mfa_level: :disabled).where.not(id: WebauthnCredential.select(:user_id)).count
```

Comment:
should this be a `distinct` on users from webauthn credentials?

Reply:
a user can have more than one WebauthN credit, right? So should we only do `where.not(id: ...)` over a distinct list of user IDs?

Reply:
I wrote a test case and didn't detect any misbehavior. Looking at the generated queries suggests why.

For example:

```ruby
webauthn_only_users = User.where(mfa_level: :disabled)
                          .where(id: WebauthnCredential.select(:user_id)).count
```

Becomes:

```sql
SELECT COUNT(*) FROM "users" 
WHERE "users"."mfa_level" = 0 
AND "users"."id" IN (
    SELECT "webauthn_credentials"."user_id" FROM "webauthn_credentials"
)
```

It's using an `IN` clause to slurp out the user IDs, rather than joining the tables. `IN` behaves like a distinct, in that only one match is needed to satisfy the clause. So it doesn't overcount.

Given that, I am not going to change the code or the tests afterall, as I think it would mean I am testing ActiveRecord instead of the business logic.